to.etc.webapp.core/src/main/java/to/etc/webapp/query/QQueryUtils.java

```diff
@@ -48,6 +48,15 @@ static public <R> R mapSelectionOneQuery(@NonNull QDataContext dc, @NonNull Clas
 		return pf.createInstance(resl);
 	}
 
+
+	public static <T extends IIdentifyable<?>, V> boolean isUnique(@NonNull QDataContext dc, Class<T> clz, @NonNull QField<T, V> property, V value, @NonNull QField<T, V> propertyId, V id ) throws Exception {
```

Comment:
No need to pass in id as separate value.
You should be able to get it with:
V id = (V) value.getId();

To avoid cast, we can also use:
PropertyMetaModel<T> pmm = MetaManager.getPropertyMeta(clz, propertyId);
V id = pmm.getValue(value);

Reply:
PropertyMetaModel and MetaManager are in to.etc.domui module...

Reply:
ok, make new class in to.etc.domui.util.db, QChecksUtil, and move the code there.
Also, fix the generic types in this, use P for id type, and then we should be able to work with less parameters over all, since both id and property value to compere can be taken from passed in value object.